From 0438372a0aeefd4d1a07b6410130317ead660526 Mon Sep 17 00:00:00 2001
From: Jason Chen <b02280@freescale.com>
Date: Sun, 19 Sep 2010 17:11:33 +0800
Subject: [PATCH 128/256] ENGR00131714 ipu lib: enc and vf task should not run
 at the same time

enc and vf task should not run at the same time, so hardware busy
check should check vf task while using enc task.

Signed-off-by: Jason Chen <b02280@freescale.com>
---
 ipu/mxc_ipu_hl_lib.c | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/ipu/mxc_ipu_hl_lib.c b/ipu/mxc_ipu_hl_lib.c
index 0911fbd..6faa3f7 100644
--- a/ipu/mxc_ipu_hl_lib.c
+++ b/ipu/mxc_ipu_hl_lib.c
@@ -372,17 +372,17 @@ static int _ipu_task_busy_in_hw(unsigned int ipu_task)
 {
 	int ret = 0;
 
-	if (ipu_task & IC_ENC)
+	if ((ipu_task & IC_ENC) ||
+		(ipu_task & IC_VF) ||
+		(ipu_task & VDI_IC_VF)) {
 		ret |= ipu_is_channel_busy(MEM_PRP_ENC_MEM);
-	if (ipu_task & IC_VF)
 		ret |= ipu_is_channel_busy(MEM_PRP_VF_MEM);
-	if (ipu_task & IC_PP)
-		ret |= ipu_is_channel_busy(MEM_PP_MEM);
-	if (ipu_task & VDI_IC_VF) {
 		ret |= ipu_is_channel_busy(MEM_VDI_PRP_VF_MEM_P);
 		ret |= ipu_is_channel_busy(MEM_VDI_PRP_VF_MEM);
 		ret |= ipu_is_channel_busy(MEM_VDI_PRP_VF_MEM_N);
 	}
+	if (ipu_task & IC_PP)
+		ret |= ipu_is_channel_busy(MEM_PP_MEM);
 	if (ipu_task & ROT_ENC)
 		ret |= ipu_is_channel_busy(MEM_ROT_ENC_MEM);
 	if (ipu_task & ROT_VF)
-- 
1.8.0

